security/src/main/java/com/fincity/security/dao/ClientUrlDAO.java

```diff
@@ -32,17 +37,31 @@ protected Field<ULong> getClientIDField() {
 	@Override
 	public Flux<ClientUrl> readAll(AbstractCondition query) {
 
-		return filter(query).flatMapMany(cond -> Mono.from(this.dslContext.select(Arrays.asList(table.fields()))
-		        .from(table)
-		        .where(cond))
-		        .map(e -> e.into(this.pojoClass)));
+		return filter(query).flatMapMany(
+				cond -> Mono.from(this.dslContext.select(Arrays.asList(table.fields())).from(table).where(cond))
+						.map(e -> e.into(this.pojoClass)));
 	}
 
 	public Mono<ClientPasswordPolicy> getClientPasswordPolicy(ULong clientId) {
 
-		return Mono.from(this.dslContext.selectFrom(SECURITY_CLIENT_PASSWORD_POLICY)
-		        .where(SECURITY_CLIENT_PASSWORD_POLICY.CLIENT_ID.eq(clientId))
-		        .limit(1))
-		        .map(e -> e.into(ClientPasswordPolicy.class));
+		return Mono
+				.from(this.dslContext.selectFrom(SECURITY_CLIENT_PASSWORD_POLICY)
+						.where(SECURITY_CLIENT_PASSWORD_POLICY.CLIENT_ID.eq(clientId)).limit(1))
+				.map(e -> e.into(ClientPasswordPolicy.class));
+	}
+
+	
+	public Mono<List<String>> getClientUrlsBasedOnAppAndClient(String appCode, ULong clientId) {
+
+		List<Condition> conds = new ArrayList<>();
+
+		conds.add(SECURITY_CLIENT_URL.APP_CODE.eq(appCode));
+
+		if (!clientId.equals(null))
```

Comment:
you should check clientId != null. If the clientId is null you can't invoke equals.